examples/src/main/java/ForEachExample.java

```diff
@@ -24,12 +25,9 @@ public static void main(String[] args) throws IOException {
             rb.add(k);
         }
         final int[] count = {0};
-        rb.forEach(new IntConsumer() {
-            @Override
-            public void accept(int value) {
-                if((value % 1500) == 0) {
-                    count[0] ++;
-                }
+        rb.forEach((IntConsumer) value -> {
```

Comment:
I don't get the need for `(IntConsumer)`

Reply:
It's an ambiguous call because `RoaringBitmap` implements both `Iterable::forEach(Consumer<? super T> action)` and  `ImmutableBitmapDataProvider::forEach(IntConsumer ic)` so it needs a cast.

Reply:
@lemire should we take this oportunity to change one of the method name? It looks a pity that a standard lambda is not applyable.

Reply:
or `implements Iterable<Integer>` should be dropped alltogether? Accepting a Consumer of autoboxed Integers looks legacy to me...

There is already org.roaringbitmap.IntIterator. We may leverage this signature change to propose a `.stream()` returning an `IntStream`.

Reply:
Dropping the `Iterable` will prevent users to use use the `RoaringBitmap` in enhanced for-Loops `(i.e., for (int i : bitmap);)`.



I actually tried `Consumer<Integer>` to remove the superfluous cast, but the trade off is syntax vs robustness. For boxed integers, nulls will be allowed (wont break compilation) and that can be problematic. You can see here: https://github.com/RoaringBitmap/RoaringBitmap/compare/master...Maithem:useJavasLongConsumer2

Reply:
> Dropping the Iterable will prevent users to use use the RoaringBitmap in enhanced for-Loops (i.e., for (int i : bitmap);).

Totally fine for me. Such could would go through an autoboxed Integers even if the `for` may suggests only primitive ints are considered.

> Consumer<Integer>

Again, I would feel this as legacy ; nulls being allowed being another sign of something wrong.

I feel we should not suggest any wrong usage of the library. Massive autoboxing looks very bad to me -> IntConsumer + IntStream are interfaces which did not exist when Iterable<Integer> was introduced.